File: Backend/WebApi/Types/EmployeeValidator.cs

using FluentValidation;

namespace WebApi.Types
{
    public class EmployeeValidator : AbstractValidator<EmployeeAggregate>
    {
        public EmployeeValidator()
        {
            RuleFor(employee => employee.name).NotEmpty();
            RuleFor(employee => employee.description).NotEmpty();
            RuleFor(employee => employee.birthday).GreaterThan(new DateTime(1900, 1, 1));
            RuleFor(employee => employee.sex).NotNull();
        }
    }
}

File: Frontend/src/app/utils/EmployeeValidator.ts
import { Validator } from 'fluentvalidation-ts';
import { Employee } from '../interfaces/Employee';

export class EmployeeValidator extends Validator<Employee> {
    constructor() {
      super();
  
      this.ruleFor('name').notEmpty()
  
        this.ruleFor('sex').notNull()

        this.ruleFor('birthday').notNull();

        this.ruleFor('description').notEmpty()
    }
  }